src/components/filters/useGroupFilter.js

```diff
@@ -31,8 +32,12 @@ export const buildHostGroupChips = (selectedGroups = []) => {
 
 const useGroupFilter = (apiParams = []) => {
     const dispatch = useDispatch();
+    const groupsEnabled = useFeatureFlag('hbi.ui.inventory-groups');
+
     useEffect(() => {
-        dispatch(fetchGroups(apiParams));
+        if (groupsEnabled === true) {
+            dispatch(fetchGroups(apiParams));
+        }
     }, []);
```

Comment:
Do you think it would make sense to add the `groupsEnabled` as a dependency here?

Reply:
Good catch, will do